airbyte-integrations/connectors/source-recharge/source_recharge/api.py

```diff
@@ -145,10 +144,23 @@ class Metafields(RechargeStream):
     Metafields Stream: https://developer.rechargepayments.com/v1-shopify?python#list-metafields
     """
 
-    def read_records(self, stream_slice: Optional[Mapping[str, Any]] = None, **kwargs) -> Iterable[Mapping[str, Any]]:
+    def request_params(
+            self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None,
+            **kwargs
+    ) -> MutableMapping[str, Any]:
+        params = {"limit": self.limit, "owner_resource": (stream_slice or {}).get("owner_resource")}
+        if next_page_token:
+            params.update(next_page_token)
+
+        return params
+
+    def stream_slices(
+            self, sync_mode: SyncMode, cursor_field: List[str] = None, stream_state: Mapping[str, Any] = None
+    ) -> Iterable[Optional[Mapping[str, Any]]]:
         owner_resources = ["customer", "store", "subscription"]
+
         for owner in owner_resources:
-            yield from super().read_records(stream_slice={"owner_resource": owner}, **kwargs)
+            yield {"owner_resource": owner}
```

Comment:
nit, but we could avoid the loop and create the slices and yield in fewer lines:
```
owner_resources = ["customer", "store", "subscription"]
yield from [{"owner_resource": owner} for owner in owner_resources]
```